File: question-service/src/routes/question-routes.ts
// import { Router } from 'express';
// import { templateController } from '../controller/template-controller';

// const router = Router();

// router.get('/', templateController.home);

// export default router;

import { Router } from "express";
import { questionController } from "../controller/question-controller";
import authMiddleware from "../middleware/question-middleware";
import upload from "../middleware/image-upload";

const router = Router();

// Route to create a question
router.post(
  "/",
  //  upload.array("images", 5),
  questionController.createQuestion
);

// Route to get all questions
router.get("/", questionController.getAllQuestions);

// Route to get a question by ID
router.get("/:id", questionController.getQuestionById);

// Route to update a question
router.put(
  "/:id",
  //   authMiddleware,
  //   upload.array("images", 5),
  questionController.updateQuestion
);

// Route to delete a question
router.delete(
  "/:id",
  // authMiddleware,
  questionController.deleteQuestion
);

export default router;

File: backend/question-service/internal/routes/questionRoutes.go
package routes

import (
	"net/http"

	"question-service/internal/controllers"
	"question-service/internal/middleware"

	"github.com/gorilla/mux"
)

// RegisterQuestionRoutes defines the API routes for managing questions
func RegisterQuestionRoutes(router *mux.Router) {
	// Public routes (GET requests) - Require authentication
	router.Handle("/questions", http.HandlerFunc(controllers.GetAllQuestions)).Methods("GET")

	// Route for collaboration service to get a matching question based on complexity and category
	router.Handle("/questions/get-question", http.HandlerFunc(controllers.GetQuestion)).Methods("GET")

	router.Handle("/questions/{id}", http.HandlerFunc(controllers.GetQuestionByID)).Methods("GET")

	// Admin-only routes (POST, PUT, DELETE requests) - Require both authentication and admin privileges
	router.Handle("/questions", middleware.VerifyAccessToken(middleware.ProtectAdmin(http.HandlerFunc(controllers.CreateQuestion)))).Methods("POST")
	router.Handle("/questions/{id}", middleware.VerifyAccessToken(middleware.ProtectAdmin(http.HandlerFunc(controllers.UpdateQuestion)))).Methods("PUT")
	router.Handle("/questions/{id}", middleware.VerifyAccessToken(middleware.ProtectAdmin(http.HandlerFunc(controllers.DeleteQuestion)))).Methods("DELETE")
}
